solr/contrib/ltr/src/java/org/apache/solr/ltr/search/LTRQParserPlugin.java

```diff
@@ -146,93 +149,114 @@ public LTRQParser(String qstr, SolrParams localParams, SolrParams params,
     @Override
     public Query parse() throws SyntaxError {
```

Comment:
7/n minor comments:
* technically not just `modelNames[0]` but all the model names could be `isEmpty` checked
* `threadManager.setExecutor` (already mentioned in 6/n) need not be inside the loop, likewise `extractFeatures` and `fvStoreName` and `extractEFIParams(localParams)` could move out.

Reply:
as you will see in my next commit, I added the empty check in the loop and a couple of tests for it, I leave this open, feel free to resolve this once you take a look.

If we don't check for empty, we will get an exception anyway saying:
"cannot find model"
It's up to us if we want to help more the user with the empty message.
I think is fine, telling explicitly in an exception the model is empty could save time for developers/integrators not understanding why a model cannot be found